packages/ava/src/advisor/advise-pipeline/advicesForChart.ts

```diff
@@ -4,69 +4,90 @@ import { cloneDeep } from '../utils';
 import { dataToAdvices } from './data-to-advices';
 
 import type { CkbTypes } from '../../ckb';
+import type { Data } from '../../common/types';
 import type { AdviseResult, ChartAdviseParams } from '../types';
 import type { BasicDataPropertyForAdvice, RuleModule } from '../ruler';
 
+export function getDataProps(
+  copyData: Data,
+  fields: string[],
+  dataProps?: Partial<BasicDataPropertyForAdvice>[]
+): BasicDataPropertyForAdvice[] {
+  // transform data into DataFrame
+  let dataFrame: DataFrame;
+  if (fields) {
+    dataFrame = new DataFrame(copyData, { columns: fields });
+  } else {
+    dataFrame = new DataFrame(copyData);
+  }
+  let dataPropsForAdvice: BasicDataPropertyForAdvice[];
+  const defaultDataProps = dataFrame.info() as BasicDataPropertyForAdvice[];
+  if (dataProps) {
+    // filter out fields that are not included for advising
+    const inputDataProps = fields
+      ? dataProps.filter((dataProp: BasicDataPropertyForAdvice) => fields.includes(dataProp.name))
+      : dataProps;
+    dataPropsForAdvice = inputDataProps.map((item) => {
+      const dwProps = defaultDataProps.find((dwItem) => {
+        return item.name === dwItem.name;
+      });
+      return {
+        ...dwProps,
+        ...item,
+      };
+    });
+  } else {
+    dataPropsForAdvice = defaultDataProps;
+  }
```

Comment:
since you had already filtered fields in `new DataFrame(copyData, { columns: fields })`, the dataFrame is filtered and you might not need to do `inputDataProps = fields ? : ;` again.

however, is this the right way to use new DataFrame()? 
what if { columns } would made the DataFrame lose some info in the future (like relation between fields)?
@pddpd

Reply:
> since you had already filtered fields in `new DataFrame(copyData, { columns: fields })`, the dataFrame is filtered and you might not need to do `inputDataProps = fields ? : ;` again.
> 
> however, is this the right way to use new DataFrame()? what if { columns } would made the DataFrame lose some info in the future (like relation between fields)? @pddpd

`inputDataProps  =  dataProps` is fine. @GuangMingYouBei 
If set `{columns: fields}`,  DF will filter original data and output data with specified columns. If users need original data information, he should use original data.

Reply:
done